File: client_angular/src/app/models/ligne-commande.ts
// ligne-commande.model.ts
export interface LigneCommande {
  id?: number;
  pieceId: number;
  pieceNom?: string;
  prixUnitaire?: number;
  quantite: number;
  livraison?: string; // 'standard' | 'express'
  total?: number;
}

File: springboot/app/src/main/java/com/bugshan/automative/group/app/model/LigneCommande.java
package com.bugshan.automative.group.app.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LigneCommande {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int quantite;

    private String livraison; // standard, express...

    private double prixUnitaire; // prix figé au moment de la commande

    @ManyToOne
    @JoinColumn(name = "piece_id")
    private Piece piece;

    @ManyToOne
    @JoinColumn(name = "commande_id")
    private Commande commande;
}
